engine/grid.py

```diff
@@ -211,26 +222,49 @@ def activate_rectangle(self, row, col, row_limit, col_limit):
         Activates all the nodes in a rectangle based on row start/end, col
         start/end.
         """
+        self.horizontal_waypoint_list = [[] for _ in range(col, col_limit)]
```

Comment:
is col_limit - col = num_col?

Reply:
I think? I was following the logic in the original code